### PR TITLE
[BUG]Add downgrade logic when fetching indexes 

### PR DESCRIPTION
**Pull request by SuZhou-Joe**
### Description
[Describe what this change achieves]

### Issues Resolved
#638 
### Check List
- [x] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Review by gaobinlong — APPROVED** [review]
LGTM.

File: server/services/IndexService.ts [hunk 1]
```diff
@@ -77,12 +77,12 @@ export default class IndexService {
         callWithRequest("cat.recovery", {
           format: "json",
           detailed: true,
-        }),
+        }).catch(() => []),
```
Comment (Hailong-am) [inline server/services/IndexService.ts:80]: should we only fix for missing permission case (http status code 403) or all exception case

**Review by Hailong-am — APPROVED** [review]
LGTM

File: server/services/IndexService.ts (hunk 1 above)
Comment (SuZhou-Joe) [reply server/services/IndexService.ts:80]: Actually no matter what error cat.recovery API gives we should wrap and gives the raw response the cat.indexes API gives.

**Pull request merged**